src/core/providers/arcgis/qgsarcgisrestutils.cpp

```diff
@@ -374,15 +374,26 @@ QgsAbstractGeometry *QgsArcGisRestUtils::convertGeometry( const QVariantMap &geo
 
 QgsCoordinateReferenceSystem QgsArcGisRestUtils::convertSpatialReference( const QVariantMap &spatialReferenceMap )
 {
+  QgsCoordinateReferenceSystem crs;
+
   QString spatialReference = spatialReferenceMap[QStringLiteral( "latestWkid" )].toString();
   if ( spatialReference.isEmpty() )
     spatialReference = spatialReferenceMap[QStringLiteral( "wkid" )].toString();
-  if ( spatialReference.isEmpty() )
-    spatialReference = spatialReferenceMap[QStringLiteral( "wkt" )].toString();
+
+  if ( spatialReference.isEmpty() && !spatialReferenceMap[QStringLiteral( "wkt" )].toString().isEmpty() )
+  {
+    crs.createFromWkt( spatialReferenceMap[QStringLiteral( "wkt" )].toString() );
+  }
   else
```

Comment:
```suggestion
  else if ( !spatialReference.isEmpty() )
```

Just a small suggestion to make the code a tiny bit easier to read of first approach. Ignore at will.

Reply:
thanks, I've flipped the condition to make this whole function easier to read